fastai/optimizer.py

```diff
@@ -11,12 +11,24 @@
 # Cell
 class _BaseOptimizer():
     "Common functionality between `Optimizer` and `OptimWrapper`"
-    def all_params(self, n=slice(None), with_grad=False):
+    def all_params(self,
+        n=slice(None), # Where to slice the model or all as default
+        with_grad=False # Get all param tuples. If `True` select only those with a gradient
```

Comment:
By param tuples, you are referring to `(p,pg,self.state[p],hyper)` , right? I don't know if param tuples is self-explanatory. Maybe say something like the parameters, states, hyperparams, etc.

Reply:
Yeah, the thing is that we dont have a "concept" (or enum/class/dataclass) to call this combination of things or tuple, it is only transient/needed in the scope of this methods.